### DISCUSSION TITLE
Chain of validation rules does not work correctly

### DISCUSSION THREAD

**Discussion by TheAndrey** [laravel/framework#47962] [Ideas]
### Laravel Version

10.16.1

### PHP Version

8.1.21

### Description

Hi,
This is not the first time I have encountered unsatisfactory validation work.

I changed the name of the current password entry field as follows, which anyone can do using the developer tools in the browser:

```html
<input name="current_password[]" type="password" autocomplete="current-password" required />
```

I use the following validation rule:

```php
$request->validate([
    'current_password' => ['required', 'string', 'current_password']
]);
```

When submitting a form, an error occurs in the `current_password` rule, despite the fact that it is preceded by a rule prohibiting accepting an array.

```
TypeError: password_verify(): Argument #1 ($password) must be of type string, array given in file vendor/laravel/framework/src/Illuminate/Hashing/AbstractHasher.php on line 32

#0 vendor/laravel/framework/src/Illuminate/Hashing/AbstractHasher.php(32): password_verify()
#1 vendor/laravel/framework/src/Illuminate/Hashing/BcryptHasher.php(74): Illuminate\Hashing\AbstractHasher->check()
#2 vendor/laravel/framework/src/Illuminate/Hashing/HashManager.php(76): Illuminate\Hashing\BcryptHasher->check()
#3 vendor/laravel/framework/src/Illuminate/Validation/Concerns/ValidatesAttributes.php(509): Illuminate\Hashing\HashManager->check()
#4 vendor/laravel/framework/src/Illuminate/Validation/Validator.php(624): Illuminate\Validation\Validator->validateCurrentPassword()
#5 vendor/laravel/framework/src/Illuminate/Validation/Validator.php(430): Illuminate\Validation\Validator->validateAttribute()
#6 vendor/laravel/framework/src/Illuminate/Validation/Validator.php(461): Illuminate\Validation\Validator->passes()
#7 vendor/laravel/framework/src/Illuminate/Validation/Validator.php(503): Illuminate\Validation\Validator->fails()
#8 vendor/laravel/framework/src/Illuminate/Foundation/Providers/FoundationServiceProvider.php(122): Illuminate\Validation\Validator->validate()
#9 vendor/laravel/framework/src/Illuminate/Macroable/Traits/Macroable.php(123): Illuminate\Http\Request->Illuminate\Foundation\Providers\{closure}()
#10 app/Http/Controllers/IndexController.php(237): Illuminate/Http/Request->__call()
```

# My suggestion

Processing of the chain of rules (from left to right) of each field should be interrupted as soon as one of the rules returns an error.
I am faced with this problem every time I write my rules - incorrect input comes into them, despite the presence of rules prohibiting it in front of him. I have to write extra logic for checking the received value types.

Processing input arrays is a Laravel sore spot... 😵
#47913

### Steps To Reproduce

Modify the password input field so that it sends an array.

**Comment by timacdonald** [comment]
Hi @TheAndrey, you will want to take a look at the [`bail` validation rule](
https://laravel.com/docs/10.x/validation#rule-bail) to stop on a input-by-input basis.

If you are using a form request, you may use the [`stopOnFirstFailure` property](https://laravel.com/docs/10.x/validation#request-stopping-on-first-validation-rule-failure), if you want this across all validation.

If you are manually creating a validator, you can call the [`stopOnFirstFailure()` method](https://laravel.com/docs/10.x/validation#manually-creating-validators) to apply it across all validation.
